src/runners/scilla_runner.ml

```diff
@@ -278,22 +297,48 @@ let () =
         in
         let m = Msg mmsg in
 
-        (* Retrieve state variables *)
-        let (curargs, cur_bal) = 
-        try
-          input_state_json cli.input_state
-        with
-        | Invalid_json s ->
-            fatal_error_gas
-              (s @ (mk_error0 (sprintf "Failed to parse json %s:\n" cli.input_state)))
-            gas_remaining
+        let cstate, gas_remaining' = 
+        if is_ipc then
+          (* TODO: Get balance as command line argument. *)
+          let cur_bal = Stdint.Uint128.zero in
+          let init_res = init_module cmod initargs [] cur_bal bstate elibs in
+          let cstate, gas_remaining', _ = check_extract_cstate cli.input init_res gas_remaining in
+          (* Initialize the state server. *)
+          let fields = List.map cstate.fields ~f:(fun (s, t) ->
+            let open StateService in
+            { fname = s; ftyp = t; fval = None }
+          ) in
+          let () = StateService.initialize ~sm:(IPC cli.ipc_port) ~fields in
+          (cstate, gas_remaining')
+        else
+
+          (* Retrieve state variables *)
+          let (curargs, cur_bal) = 
+          try
+            input_state_json cli.input_state
+          with
+          | Invalid_json s ->
+              fatal_error_gas
+                (s @ (mk_error0 (sprintf "Failed to parse json %s:\n" cli.input_state)))
+              gas_remaining
+          in
+
+          (* Initializing the contract's state *)
+          let init_res = init_module cmod initargs curargs cur_bal bstate elibs in
+          (* Prints stats after the initialization and returns the initial state *)
+          (* Will throw an exception if unsuccessful. *)
+          let cstate, gas_remaining', field_vals = check_extract_cstate cli.input init_res gas_remaining in
+
+          (* Initialize the state server. *)
+          let fields = List.map field_vals ~f:(fun (s, l) ->
```

Comment:
So is it the same as before, just moving the state-updating logic into a separate module?